src/EditorFeatures/Test2/FindReferences/FindReferencesTests.AnonymousTypeSymbols.vb

```diff
@@ -165,5 +166,57 @@ End Class
 </Workspace>
             Await TestAPIAndFeature(input, kind, host)
         End Function
+#End Region
+
+#Region "C#"
+        <WorkItem("https://github.com/dotnet/roslyn/issues/20115")>
+        <WpfTheory, CombinatorialData>
+        Public Async Function TestAnonymousTypesCSharp_SimpleExpression(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+            <![CDATA[
+class C
+{
+    void M(string {|Definition:a|})
+    {
+        int length = [|a|].Length;
+        var r = new { [|a|], Length = 1 };
+        r = new { $$[|a|], [|a|].Length };
+    }
+}
+]]>
+        </Document>
+    </Project>
+</Workspace>
+            Await TestAPIAndFeature(input, kind, host)
+        End Function
+
+        <WorkItem("https://github.com/dotnet/roslyn/issues/20115")>
+        <WpfTheory, CombinatorialData>
+        Public Async Function TestAnonymousTypesCSharp_PropertyAccess(kind As TestKind, host As TestHost) As Task
+            Dim input =
+<Workspace>
+    <Project Language="C#" CommonReferences="true">
+        <Document>
+            <![CDATA[
+class C
+{
+    void M(string a)
+    {
+        int length = a.[|Length|];
+        var r = new { a, Length = 1 };
+        r = new { a, a.$$[|Length|] };
+        r = new { a, Length = a.[|Length|] };
```

Comment:
this slightly confuses me.  shouldn't this be `[|Length|] = a.[|Length|]`?   If not, why not?

Reply:
The logic that made the most sense to me is to not consider the name of the explicit name of the anon type since it's a property of `string`, and we only refer to the property of `string`. The `Length` property that is explicitly named is a different symbol that happens to have the same name as `string.Length`. Hence this case.

Reply:
But the line above is `r = new { a, a.Length };` right? So that will have cascaded the property name to the name of the anonymous type member.

Reply:
Right, but since we're accessing `a`, the user is more likely to be referring to the specific property, rather than also including all the properties that have their name implied from that one. In which case, you're also highlighting a property for another type which isn't directly related to the property of string, but any property named Length.

If that's too confusing, the point is that it made sense to me to exclude the definition of the anonymous type. Though I'm open to whatever suits best for this scenario.

Reply:
This is the purpose of cascading. 

Note: this is also needed of we expect rename to work. Which is a prime case for find-refs. 

> Right, but since we're accessing a, the user is more likely to be referring to the specific property

This is the API level though. The API will return the two different symbols, and there references between each. 

The UI/feature level can choose if it wants to filter things out or not. We have the concept of FindRef options for that (though I'm skeptical we would need that here)

Reply:
So the rationale is we return both symbols because we highlight the property whose name implies the name of the anonymous type property, but we do not return all other properties named with the same name that are not assignable to a variable of the original anonymous type. For example,

```csharp
string s;
var a = new { A = 1, s.Length };
a = new { A = 1, Length = 2 }; // we highlight Length here
a = new { B = 1, Length = 2 }; // we do not highlight Length here
```

In that sense we have this sort of "dual view" that the one issue mentions, if I understand correctly.

Reply:
Correct